### PR TITLE
feat: commit template for new component contributions

### PR DESCRIPTION
**Pull request by benlister-okta**
Adds Github commit template for new component contributions.

[OKTA-672078](https://oktainc.atlassian.net/browse/OKTA-672078)
https://oktainc.atlassian.net/browse/OKTA-672078

## Summary

Creating a new template that we can use for new component contributions. The template includes the Developer checklist from the Odyssey Component Workflow project. The workflow for using this component will be integrated separately after this template is merged.

### CODE REVIEW

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+### Internationalization (i18n) implementation
+
+- [ ] Separate text content from the component's code for translation.
+- [ ] Ensure the component renders correctly when language changes.
+
+### Responsive implementation
+
+- [ ] Use CSS media queries to define breakpoints and apply styles.
+- [ ] Test fluid layouts and resizing behavior at each breakpoint.
+- [ ] Optimize images for different screen resolutions.
```
Comment (bryancunningham-okta) [inline .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:46]: I know this is part of the component workflow checklist but, now that I think about it, we probably don't need this. Odyssey shouldn't be serving any images, I wouldn't think

Comment (benlister-okta) [reply .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md]: @bryancunningham-okta makes sense, I updated the text to say "as needed". The comment in the template also mentions to use of relevant parts. Do you think that would help avoid confusion?

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+- [ ] Use CSS media queries to define breakpoints and apply styles.
+- [ ] Test fluid layouts and resizing behavior at each breakpoint as needed.
+- [ ] Optimize images for different screen resolutions as needed.
+
+### Cross-browser compatibility
+
+- [ ] Test the component on major browsers (Chrome, Firefox, Safari, Edge, etc.).
+- [ ] Use browser prefixes or CSS libraries to handle compatibility issues.
+- [ ] Ensure consistent behavior and appearance across browsers.
```
Comment (KevinGhadyani-Okta) [inline .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:52]: When you add these checkboxes, no PR can get merged unless they're all checked.

Is that our intention?

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+- [ ] Provide visual alternatives for auditory elements.
+
+### Internationalization (i18n) implementation
+
+- [ ] Separate text content from the component's code for translation.
+- [ ] Ensure the component renders correctly when language changes.
+
+### Responsive implementation
+
+- [ ] Use CSS media queries to define breakpoints and apply styles.
```
[the remark this answers is not in the document]
Comment (KevinGhadyani-Okta) [reply .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:44]: Ah, I see. This is a different PR template.

Is there a way we can ensure user's have to select one of these when creating a PR?

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+### QA
+
+- [ ] Conduct unit testing to continuously verify individual component functionality.
+- [ ] Include unit tests to continuously verify component functionality
+- [ ] Screen reader test error/success states or message
+- [ ] Test edge cases and error scenarios to validate robustness
+
+### Testing and validation
+
+- [ ] Work shared in [Craft](https://docs.google.com/document/d/1dqaEzKW3SlKccw2ILjX7y3LQY0toeNe9OuwpOs2XxcI/edit#heading=h.j8zakyclqze5) to get feedback from the Odyssey team
```
Comment (KevinGhadyani-Okta) [inline .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:78]:
```suggestion
- [ ] Share work in [Craft](https://docs.google.com/document/d/1dqaEzKW3SlKccw2ILjX7y3LQY0toeNe9OuwpOs2XxcI/edit#heading=h.j8zakyclqze5) to get feedback from the Odyssey team
```

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (benlister-okta) [reply .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:44]: Yes, this would be a new template, I was thinking there are a few ways we could implement it: Links in the main template (e.g., use this for new components), sharing a link we can generate from Github (I found a how-to on this), or possibly pick it dynamically based on the commit message type/content or Github label.

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (benlister-okta) [reply .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:52]: My thought was that people could remove irrelevant items from the list to avoid that. I had a comment at the top of the template that was probably a too subtle, I just updated it to say:
`<!-- Pease track and mark relevant items as part of the component development process. Remove any non-applicable items. -->`

File: .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+### Documentation
+
+- [ ] Create clear and concise usage documentation with step-by-step instructions.
+- [ ] Include code snippets, demos, and examples for various use cases.
+- [ ] Provide guidance on customization and handling special cases.
+
+### Finalize code
+
+- [ ] Review PR for final detailing
+- [ ] Approve PR in GitHub
```
[the remark this answers is not in the document]
Comment (benlister-okta) [reply .github/NEW_COMPONENT_PULL_REQUEST_TEMPLATE.md:63]: Good point, I removed them.

**Review by KevinGhadyani-Okta — APPROVED** [review]
